quic/s2n-quic-core/src/datagram.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+pub trait DatagramApi: Sender + Receiver {}
```

Comment:
I would call this `Endpoint` since it'll live on the endpoint. Then eventually it'll return instances when connections are created. That being said, this trait doesn't implement the Sender or Receiver, but instead creates and returns them for each connection.

Reply:
yeah something like:

```rust
pub trait Endpoint: 'static + Send {
    type Sender: Sender;
    type Receiver: Receiver;

    fn create_connection(&mut self, info: &ConnectionInfo) -> (Self::Sender, Self::Receiver);
}
```